### DISCUSSION TITLE
Anny time improvement sollution?

### DISCUSSION THREAD

**Discussion by taoufikbn** [JanDeDobbeleer/oh-my-posh#2250] [Troubleshoot] [answered]
"No Idea what makes the exe time that slow , and how to solve it "
"Thanks for the answer in advance "

oh-my-posh debug                                                                               pwsh   100  22:13:57 

Version: 7.81.1

Segments:

ConsoleTitle(true)   -   0 ms - pwsh in ~
session(true)        -   1 ms -  taouf 
path(true)           -   0 ms -    ~
git(false)           -   9 ms -
node(false)          -   0 ms -
go(false)            -   0 ms -
julia(false)         -   0 ms -
python(false)        -   0 ms -
ruby(false)          -   0 ms -
azfunc(false)        -   0 ms -
aws(false)           -   0 ms -
root(false)          -   0 ms -
executiontime(true)  -   0 ms -  ﮫ0ms⠀
exit(true)           -   1 ms -   
shell(true)          -   0 ms -   pwsh 
ytm(false)           -  24 ms -
battery(true)        -   3 ms -   100
time(true)           -   0 ms -  22:13:59 

Run duration: 60.7988ms

Cache path: C:\Users\taouf\AppData\Local\oh-my-posh

Config path: C:\Users\taouf\AppData\Local\Programs\oh-my-posh\themes\jandedobbeleer.omp.json

Logs:

2022/05/10 22:13:58 debug: Getenv
C:\Users\taouf\AppData\Local
2022/05/10 22:13:58 Getenv duration: 83.9µs, args: LOCALAPPDATA
2022/05/10 22:13:58 CachePath duration: 253.6µs, args:
2022/05/10 22:13:58 debug: Getenv
C:\Users\taouf\AppData\Local\Programs\oh-my-posh\themes\jandedobbeleer.omp.json
2022/05/10 22:13:58 Getenv duration: 0s, args: POSH_THEME
2022/05/10 22:13:59 Shell duration: 15.4813ms, args:
2022/05/10 22:13:59 resolveConfigPath duration: 15.4813ms, args:
2022/05/10 22:13:59 Init duration: 15.8249ms, args:
2022/05/10 22:13:59 Flags duration: 0s, args:
2022/05/10 22:13:59 config.loadConfig duration: 1.0044ms, args:
2022/05/10 22:13:59 Flags duration: 0s, args:
2022/05/10 22:13:59 debug: Getenv

2022/05/10 22:13:59 Getenv duration: 0s, args: OMP_CACHE_DISABLED
2022/05/10 22:13:59 Root duration: 0s, args:
2022/05/10 22:13:59 Shell duration: 0s, args:
2022/05/10 22:13:59 ErrorCode duration: 0s, args:
2022/05/10 22:13:59 IsWsl duration: 0s, args:
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 PathSeparator duration: 0s, args:
2022/05/10 22:13:59 PathSeparator duration: 0s, args:
2022/05/10 22:13:59 debug: User
taouf
2022/05/10 22:13:59 User duration: 0s, args:
2022/05/10 22:13:59 debug: Host
LAPTOP-ONBBD2J4
2022/05/10 22:13:59 Host duration: 0s, args:
2022/05/10 22:13:59 GOOS duration: 0s, args:
2022/05/10 22:13:59 TemplateCache duration: 0s, args:
2022/05/10 22:13:59 Flags duration: 0s, args:
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 GOOS duration: 0s, args:
2022/05/10 22:13:59 debug: Getenv

2022/05/10 22:13:59 Getenv duration: 0s, args: SSH_CONNECTION
2022/05/10 22:13:59 debug: Getenv

2022/05/10 22:13:59 Getenv duration: 0s, args: SSH_CLIENT
2022/05/10 22:13:59 TemplateCache duration: 0s, args:
2022/05/10 22:13:59 TemplateCache duration: 0s, args:
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 GOOS duration: 0s, args:
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 Flags duration: 0s, args:
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 GOOS duration: 0s, args:
2022/05/10 22:13:59 GOOS duration: 0s, args:
2022/05/10 22:13:59 PathSeparator duration: 0s, args:
2022/05/10 22:13:59 PathSeparator duration: 0s, args:
2022/05/10 22:13:59 PathSeparator duration: 0s, args:
2022/05/10 22:13:59 GOOS duration: 0s, args:
2022/05/10 22:13:59 IsWsl duration: 0s, args:
2022/05/10 22:13:59 StackCount duration: 0s, args:
2022/05/10 22:13:59 TemplateCache duration: 0s, args:
2022/05/10 22:13:59 TemplateCache duration: 0s, args:
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 GOOS duration: 0s, args:
2022/05/10 22:13:59 GOOS duration: 0s, args:
2022/05/10 22:13:59 HasCommand duration: 9.7947ms, args: git.exe
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 error: HasParentFilePath
CreateFile C:\.git: The system cannot find the file specified.
2022/05/10 22:13:59 HasParentFilePath duration: 0s, args: .git
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 GOOS duration: 0s, args:
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 GOOS duration: 0s, args:
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 GOOS duration: 0s, args:
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 GOOS duration: 0s, args:
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 PathSeparator duration: 0s, args:
2022/05/10 22:13:59 debug: HasFiles
false
2022/05/10 22:13:59 HasFiles duration: 998.3µs, args: *.py
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 PathSeparator duration: 0s, args:
2022/05/10 22:13:59 debug: HasFiles
false
2022/05/10 22:13:59 HasFiles duration: 0s, args: *.ipynb
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 PathSeparator duration: 0s, args:
2022/05/10 22:13:59 debug: HasFiles
false
2022/05/10 22:13:59 HasFiles duration: 0s, args: pyproject.toml
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 PathSeparator duration: 0s, args:
2022/05/10 22:13:59 debug: HasFiles
false
2022/05/10 22:13:59 HasFiles duration: 0s, args: venv.bak
2022/05/10 22:13:59 debug: HasFolder
false
2022/05/10 22:13:59 HasFolder duration: 0s, args: .venv
2022/05/10 22:13:59 debug: HasFolder
false
2022/05/10 22:13:59 HasFolder duration: 0s, args: venv
2022/05/10 22:13:59 debug: HasFolder
false
2022/05/10 22:13:59 HasFolder duration: 0s, args: virtualenv
2022/05/10 22:13:59 debug: HasFolder
false
2022/05/10 22:13:59 HasFolder duration: 0s, args: env
2022/05/10 22:13:59 debug: HasFolder
false
2022/05/10 22:13:59 HasFolder duration: 0s, args: venv-win
2022/05/10 22:13:59 debug: HasFolder
false
2022/05/10 22:13:59 HasFolder duration: 0s, args: pyenv-win
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 GOOS duration: 0s, args:
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 GOOS duration: 0s, args:
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 GOOS duration: 0s, args:
2022/05/10 22:13:59 debug: Getenv

2022/05/10 22:13:59 Getenv duration: 0s, args: AWS_VAULT
2022/05/10 22:13:59 debug: Getenv

2022/05/10 22:13:59 Getenv duration: 0s, args: AWS_PROFILE
2022/05/10 22:13:59 debug: Getenv

2022/05/10 22:13:59 Getenv duration: 0s, args: AWS_REGION
2022/05/10 22:13:59 debug: Getenv

2022/05/10 22:13:59 Getenv duration: 0s, args: AWS_DEFAULT_REGION
2022/05/10 22:13:59 debug: Getenv

2022/05/10 22:13:59 Getenv duration: 0s, args: AWS_CONFIG_FILE
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 error: FileContent
open C:\Users\taouf/.aws/config: The system cannot find the path specified.
2022/05/10 22:13:59 FileContent duration: 0s, args: C:\Users\taouf/.aws/config
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 GOOS duration: 0s, args:
2022/05/10 22:13:59 Root duration: 0s, args:
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 GOOS duration: 0s, args:
2022/05/10 22:13:59 ExecutionTime duration: 0s, args:
2022/05/10 22:13:59 TemplateCache duration: 0s, args:
2022/05/10 22:13:59 TemplateCache duration: 0s, args:
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 GOOS duration: 0s, args:
2022/05/10 22:13:59 ErrorCode duration: 0s, args:
2022/05/10 22:13:59 TemplateCache duration: 0s, args:
2022/05/10 22:13:59 TemplateCache duration: 0s, args:
2022/05/10 22:13:59 TemplateCache duration: 0s, args:
2022/05/10 22:13:59 Flags duration: 0s, args:
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 GOOS duration: 0s, args:
2022/05/10 22:13:59 Shell duration: 0s, args:
2022/05/10 22:13:59 Flags duration: 0s, args:
2022/05/10 22:13:59 TemplateCache duration: 0s, args:
2022/05/10 22:13:59 TemplateCache duration: 0s, args:
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 GOOS duration: 0s, args:
2022/05/10 22:13:59 error: HTTPRequest
Get "http://127.0.0.1:9863/query": context deadline exceeded
2022/05/10 22:13:59 HTTPRequest duration: 24.0046ms, args: http://127.0.0.1:9863/query
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 GOOS duration: 0s, args:
2022/05/10 22:13:59 IsWsl duration: 0s, args:
2022/05/10 22:13:59 BatteryInfo duration: 3.0007ms, args:
2022/05/10 22:13:59 TemplateCache duration: 0s, args:
2022/05/10 22:13:59 TemplateCache duration: 0s, args:
2022/05/10 22:13:59 TemplateCache duration: 0s, args:
2022/05/10 22:13:59 TemplateCache duration: 0s, args:
2022/05/10 22:13:59 debug: Pwd
C:\Users\taouf
2022/05/10 22:13:59 Pwd duration: 0s, args:
2022/05/10 22:13:59 debug: Home
C:\Users\taouf
2022/05/10 22:13:59 GOOS duration: 0s, args:
2022/05/10 22:13:59 TemplateCache duration: 0s, args:
2022/05/10 22:13:59 TemplateCache duration: 0s, args:
2022/05/10 22:13:59 debug: Getenv
C:\Users\taouf\AppData\Local
2022/05/10 22:13:59 Getenv duration: 0s, args: LOCALAPPDATA
2022/05/10 22:13:59 CachePath duration: 0s, args:
2022/05/10 22:13:59 Flags duration: 0s, args:

**Comment by JanDeDobbeleer** [comment, marked as answer]
Hey, looking at that oh-my-posh itself takes 60ms, the problem most likely originates from Defender or any other malware/virus scanner blocking the execution. Adding am exclusion will 9/10 solve the issue.
